### PR TITLE
support array

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -150,6 +150,116 @@ export function literal2Asm(l: string): [string, string] {
   throw new Error(`<${l}> cannot be cast to ASM format, only sCrypt native types supported`);
 }
 
+
+
+
+/**
+ * convert literals to Scrypt Type
```
Comment (xhliu) [inline src/utils.ts:157]: Duplicate of https://github.com/scrypt-sv/scryptlib/blob/d6526fa03d7ed2acf14e3e146c545eb2cbc5b1f7/src/utils.ts#L58

**Review by xhliu — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 1 above)
Comment (zhfnjust) [reply src/utils.ts]: add a new function literalParser.

**Review by xhliu — CHANGES_REQUESTED** [review]
This works 👍 

Let's simplify it.

File: src/utils.ts [hunk 2]
```diff
@@ -42,114 +42,206 @@ export function int2Asm(str: string): string {
...
+  RIPEMD160 = 'Ripemd160',
+  SHA1 = 'Sha1',
+  SHA256 = 'Sha256',
+  SIGHASHTYPE = 'SigHashType',
+  SIGHASHPREIMAGE = 'SigHashPreimage',
+  OPCODETYPE = 'OpCodeType'
+}
+
+
+export function parseLiteral(l: string): [string /*asm*/ , ValueType /*value*/, VariableType /*VariableType*/] {
```
Comment (xhliu) [inline src/utils.ts:96]: `export function parseLiteral(l: string): [string /*asm*/ , ValueType, VariableType]`

File: src/utils.ts [hunk 3]
```diff
@@ -42,114 +42,206 @@ export function int2Asm(str: string): string {
...
+ * hex to bigint
  */
-export function literal2Asm(l: string): [string, string] {
+export function str2bigint(str: string):  bigint{
+  return BigInt(str);
+}
+
+export function bigint2hex(bi: bigint):  string{
+  let hex = bi.toString(16);
+  if(hex.length % 2 == 1) {
```
Comment (xhliu) [inline src/utils.ts:73]: `if(hex.length % 2 === 1) {`

**Review by xhliu — APPROVED** [review]
Good to go, well done 👍

**Pull request merged**